interface/vsomeip/internal/logger.hpp

```diff
@@ -41,12 +41,26 @@ class message
         int_type overflow(int_type);
         std::streamsize xsputn(const char *, std::streamsize);
 
+		inline void activate() {
+			active_ = true;
+		}
+
+		inline bool isActive() const {
+			return active_;
+		}
+
+		inline std::string str() const {
```

Comment:
I'm curious if a `string_view` would be more appropriate here.

Reply:
I am not 100% sure, std::stringstream::str() constructs a new string anyway and wouldn't std::string_view then point to a temporaray object? Maybe I am wrong ...

Reply:
I could be entirely wrong, but I _think_ what happens here that a new `std::string` is constructed (with allocation) here when `buffer_.str()` is involved, and then the string stream copies the contents of this into the new string it is constructing.  If this were a `std::string_stream`, we may save on one allocation.  If I'm right this could be considerable on platforms like QNX.

I suppose the only way to know would be to test it out.  I can try to set up something small on godbolt.